File: src/main/java/inclub9/intellij/fieldlabel/inspection/FieldLabelInspection.java
package inclub9.intellij.fieldlabel.inspection;

import com.intellij.codeInspection.*;
import com.intellij.psi.*;
import org.jetbrains.annotations.NotNull;

public class FieldLabelInspection extends AbstractBaseJavaLocalInspectionTool {
    @Override
    public @NotNull PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder,
                                                   boolean isOnTheFly) {
        return new JavaElementVisitor() {
            @Override
            public void visitAnnotation(@NotNull PsiAnnotation annotation) {
                if (!"FieldLabel".equals(annotation.getQualifiedName())) {
                    return;
                }

                PsiAnnotationMemberValue value = annotation.findAttributeValue("value");
                if (value == null) {
                    holder.registerProblem(annotation,
                            "FieldLabel annotation must have a value",
                            ProblemHighlightType.ERROR);
                    return;
                }

                if (!(annotation.getParent() instanceof PsiField)) {
                    holder.registerProblem(annotation,
                            "FieldLabel annotation can only be applied to fields",
                            ProblemHighlightType.ERROR);
                }
            }
        };
    }
}

File: src/main/kotlin/inclub9/intellij/plugin/fieldlabel/inspection/FieldLabelInspection.kt
package inclub9.intellij.plugin.fieldlabel.inspection

import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool
import com.intellij.codeInspection.ProblemsHolder
import com.intellij.psi.JavaElementVisitor
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.PsiField

class FieldLabelInspection : AbstractBaseJavaLocalInspectionTool() {
    
    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
        return object : JavaElementVisitor() {
            override fun visitField(field: PsiField) {
                val annotation = field.getAnnotation("inclub9.annotation.FieldLabel")
                if (annotation != null) {
                    val value = annotation.findAttributeValue("value")
                    if (value == null || (value.text.length <= 2)) { // Empty or just quotes
                        holder.registerProblem(
                            annotation,
                            "FieldLabel annotation must have a non-empty value"
                        )
                    }
                }
            }
        }
    }
}
